pmlc/dialect/pxa/transforms/autotile.h

```diff
@@ -0,0 +1,111 @@
+#pragma once
+
+#include <limits>
+#include <memory>
+#include <vector>
+
+#include "mlir/Analysis/AffineStructures.h"
+#include "mlir/Pass/Pass.h"
+#include "pmlc/dialect/pxa/ir/ops.h"
+#include "pmlc/dialect/pxa/transforms/tile.h"
+
+namespace pmlc::dialect::pxa {
+
+// A tile size generator is a functor from range -> list of sizes
+
+// Produces all powers of 2 <= range
+std::vector<int64_t> PowerOfTwoGenerator(int64_t range);
```

Comment:
nit: I'm not sure if this should be `camelCase`

Reply:
👍 Seems like most (all?) functions should be in the tightly-coupled-with-MLIR world

Reply:
Make them all functors... :)